ext/strscan/strscan.c

```diff
@@ -514,19 +514,23 @@ strscan_do_scan(VALUE self, VALUE pattern, int succptr, int getstr, int headonly
             return Qnil;
         }
         onig_region_clear(&(p->regs));
-        onig_region_set(&(p->regs), 0, 0, RSTRING_LEN(pattern));
+        onig_region_set(&(p->regs), 0, p->curr, p->curr + RSTRING_LEN(pattern));
     }
 
     MATCHED(p);
     p->prev = p->curr;
+
     if (succptr) {
-        p->curr += p->regs.end[0];
+        p->curr = p->regs.end[0];
     }
-    if (getstr) {
-        return extract_beg_len(p, p->prev, p->regs.end[0]);
-    }
-    else {
-        return INT2FIX(p->regs.end[0]);
+    {
+      long length = p->regs.end[0] - p->prev;
```

Comment:
This block of code is only indented two spaces. I'm guessing it should be four to be consistent with the code around it.

Reply:
Good catch!
Fixed.